### PR TITLE
Implement: Typing Bubble

### LINKED ISSUE #7: (BONUS) Typing... feature
We would like for users to be able to see when someone is typing. We would like you to implement the feature of "someone is typing". 

When someone in the conversation (that is not yourself) is typing, it will display the "typing bubble".

Untitled

### PR DESCRIPTION
**Pull request by michael-levitski**
## Description
Closes #7

- When a user types characters into the message input, a typing bubble will appear for the other user
- The bubble will disappear when either of the following occur:
  - The user that is typing sends the message
  - The text field is empty AND has lost focus (not sure if these are the intended conditions for the typing bubble to disappear)

## Notes on your approach and thought process
- Typing bubble logic is handled completely client-side
  - The only server-side change was initializing an `otherUserIsTyping` property to false when fetching conversations
- I created a redux action, `SET_TYPING_STATUS` that sets the `otherUserIsTyping` property for a conversation
- This action is not triggered by the user interacting with the UI. It is only triggered by a socket listener for the event `set-typing-status`
  - In other words, only the other user can trigger this action
- The `set-typing-status` socket event is emitted by:
  - Typing into the text box (sets to true)
  - The text box being empty and losing focus (sets to false)
  - Sending the message (sets to false)
 - A user performing the above 3 actions will not affect the user's own store. It will only affect the other user's store
 - The `otherUserIsTyping` property is used to decide whether the typing message bubble is rendered or not

### CODE REVIEW

**Comment by moffatethan** [comment]
Thanks @michael-levitski for doing this! Please note we'll get to this pull request between Jan 4th - Jan 7th. Happy Holidays!

File: client/src/components/ActiveChat/Input.js [hunk 1]
```diff
@@ -22,7 +23,23 @@ const Input = (props) => {
   const [text, setText] = useState("");
   const { postMessage, otherUser, conversationId, user } = props;
 
+  const emitTypingStatus = (otherUserIsTyping) => {
+    socket.emit("set-typing-status", {
+      conversationId,
+      otherUserIsTyping,
+    });
+  };
```
Comment (moffatethan) [inline client/src/components/ActiveChat/Input.js:31]: Curious on your thoughts on throttling this? If say we're typing really fast would this socket be emitting every single character typed? If we throttle we could reduce the number of emits being fired?

Probably not a huge deal for the scale of this application but if it were to have lots and lots of users typing at once it could lead to a lot of bandwidth being used.

File: client/src/components/ActiveChat/Input.js [hunk 2]
```diff
@@ -22,7 +23,23 @@ const Input = (props) => {
...
+
+  const handleBlur = () => {
+    if (!text) {
+      emitTypingStatus(false);
+    }
```
Comment (moffatethan) [inline client/src/components/ActiveChat/Input.js:36]: Nice 💯

**Review by moffatethan — APPROVED** [review]
I'm going to approve this @michael-levitski, just had one comment on a potential change but wanted to hear your thoughts on it. Otherwise, pretty clean implementation and good work on this!

File: client/src/components/ActiveChat/Input.js (hunk 1 above)
Comment (michael-levitski) [reply client/src/components/ActiveChat/Input.js:31]: I think that maybe there could be a boolean property on the conversation state such as `userIsTyping`. I could then refactor `emitTypingStatus()` to only emit the socket event if `userIsTyping` is not the same as what is passed in to the function. After the `if` block, I would set `userIsTyping` to whatever was passed to `emitTypingStatus()` as the argument.

Though I'm not sure if `userIsTyping` should be in the redux store or kept track locally in the Input component.

Comment (moffatethan) [reply client/src/components/ActiveChat/Input.js:31]: Interesting though, I do think for simplicity implementing a simple throttle or debounce would suffice for it. Though, not super important but just wanted to give you a heads up. 👍🏻 

Let's say for example the user types 150 characters, for every character we would fire a socket event. With throttling that could be reduced down to like 20 ish events or even less depending on the throttle delay set. Overall though, I like your implementation and it's clean and easy to maintain and add something like that in! 

You're good to merge this in Michael
